apps/frontend/vite.config.ts

```diff
@@ -0,0 +1,64 @@
+import { readFileSync } from "node:fs";
+import { join } from "node:path";
+import { fileURLToPath } from "node:url";
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+
+// https://vitejs.dev/config/
+export default defineConfig(({ mode }) => {
+  return {
+    mode: process.env.BUILD_MODE || mode,
+    plugins: [react()],
+    build: {
+      rollupOptions: {
+        output: {
+          manualChunks: (id) => {
+            if (id.includes("node_modules")) {
+              return "vendor";
+            }
+            return undefined;
+          },
+        },
+      },
+    },
+    resolve: {
+      alias: [
+        {
+          find: "@",
+          replacement: fileURLToPath(new URL("./src", import.meta.url)),
+        },
+      ],
+    },
+    define: {
+      "process.env.NODE_ENV": JSON.stringify(mode),
+    },
```

Comment:
Have a read: https://vitejs.dev/guide/env-and-mode 
Might be an interesting alternative to Node.js-mocking-era `process.env.*`.

Reply:
Lot of libraries still relies on process.env.NODE_ENV, I think it's better to stick on it for now. In my code not necessarily but for dependencies.